network.go

```diff
@@ -6,7 +6,10 @@ import (
 )
 
 func getHostByName(name string) string {
-	addrs, _ := net.LookupHost(name)
-	//TODO: add error handing when release v3 comes out
+	addrs, err := net.LookupHost(name)
+	if err != nil {
+		return ""
```

Comment:
it may be a good idea to return the error type (notfound vs timeout/temporary failure) as in most cases one only would want to react to a not found in the charts and abort or retry on a temporary dns resolution failure

Reply:
**Yes, that makes total sense.**

The context why I created this PR was that I was trying to use this in a dynamic Traefik config that was supposed to resolve a hostname and if that failed it would fallback to something else (resolve `host.docker.internal` for docker on Mac / Windows and fallback to `172.17.0.1`).

Because it was panicking this behavior was not possible to implement.
In the end I chose to not go with this approach anyway.

In the context of helm charts what I have implemented could be disastrous in cases like you mentioned (temporary dns resolution failure).

This PR does basically not implement error handling at all :sweat_smile: 
I would suggest I close this PR as I do not have the time to implement error handling properly + retry logic etc.